### ISSUE TITLE
Save state of game

### ISSUE THREAD

**Issue by adrian-vogler** [halogenandtoast/ArkhamHorror#747]
It would be nice to have a "Save game" and "Load game" appart from the auto-save so that one can go back to certain save games and try something different from there.

**Comment by halogenandtoast** [comment]
I think I could allow this but you would not have any undo history. Does that seem alright?Sent from my iPhoneOn Sep 28, 2024, at 11:20, adrian-vogler ***@***.***> wrote:﻿
Assigned #747 to @halogenandtoast.

**Comment by adrian-vogler** [comment]
Undo within a game is OK (but at the moment a pain if more steps involved).
Saving and loading versions of a game would be helpful

F.e. I finish part of a campaign.
Then I save this, to return to this finished campaign to try different paths from there.
So if You get stuck later You have save file from where to start and are not thrown back to the very start of the game/campaign.

Undo history would be available for new steps in the game I think, but not beyond the savepoint.
That seems reasonable.

Von: Matthew Mongeau ***@***.***>
Gesendet: 28 September 2024 12:00
An: halogenandtoast/ArkhamHorror ***@***.***>
Cc: adrian-vogler ***@***.***>; Author ***@***.***>
Betreff: Re: [halogenandtoast/ArkhamHorror] Save state of game (Issue #747)

I think I could allow this but you would not have any undo history. Does that seem alright?Sent from my iPhoneOn Sep 28, 2024, at 11:20, adrian-vogler ***@***.***<mailto:***@***.***>> wrote:﻿
Assigned #747 to @halogenandtoast.

**Comment by simonedonnini** [comment]
Yes a save feature would be great, better than undo in my opinion

**Comment by halogenandtoast** [comment]
I think I have a way where I can implement this. I am planning some updates for how you progress through the campaign and it's possible I can save snapshots at "checkpoints". Loading a checkpoint snapshot will come with the following caveats:

1) You can reset your current game to this point, however if you do so you won't be able to go back to where you came from, it will effectively be a hard reset. You'll still be able to undo the game before that checkpoint (probably not a desired feature, but it will just work out this way regardless)
2) I will allow you to spawn a copy of your game at any previous checkpoint, but you will have no undo history in this new game

I think these two are good compromises and should cover most cases for why I think people would want this feature.

Expect to see this available once I complete The Scarlet Keys.
